packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -199,6 +203,23 @@ async function coerceObject(input: string | object, spec: ParameterObject) {
   return data;
 }
 
+function coerceArray(
+  data: string | number | boolean | object,
+  spec: ParameterObject,
+) {
+  if (spec.in === 'query') {
+    if (
+      typeof data === 'string' ||
+      typeof data === 'number' ||
+      typeof data === 'boolean'
+    ) {
+      return [data];
+    }
```

Comment:
I would use the following logic:

```ts
if (data == null || Array.isArray(data)) return data; // return `null | undefined or *[]`
return [data];
```

Reply:
It's possible to accept `object[]` using stringified json:

such as `item={"x":1}&item={"x":2"}`.

My version allows  `item={"x":1}` too.

Reply:
Updated accordingly. Thanks @raymondfeng